### DISCUSSION TITLE
how to delete object with self-referenced objects

### DISCUSSION THREAD

**Discussion by pplmx** [go-gorm/gorm#6344] [Q&A]
if there is a struct like this: 
```go
type Task struct {
    ID       uint
    PrevDeps []*Task `gorm:"many2many:task_depends;joinForeignKey:NextID;joinReferences:PrevID"`
    NextDeps []*Task `gorm:"many2many:task_depends;joinForeignKey:PrevID;joinReferences:NextID"`
}
```

Using this sentence to delete, which can only delete task(ID=1) and cannot delete NextDeps
```go
var t Task
t.ID = 1
db.Select("NextDeps").Delete(&t).Error
```

**Comment by pplmx** [comment]
Could anyone give me some help? Thanks in advance. :)

**Comment by saeidee** [comment]
You can delete it using Clear & Unscope functions like the below:

<img width="707" alt="image" src="https://github.com/go-gorm/gorm/assets/29974712/9071dffd-e0f8-4cd9-89a5-76aff2cb7eb4">
